python/nav/config.py

```diff
@@ -259,6 +261,71 @@ def _install_single_config_resource_(source, target, overwrite=False):
         return target_file
 
 
+def verify_nav_config(config):
+    """Verifies the validity of the most critical options of nav.conf
+
+    :param config: A NAV config dictionary, typically the result of
+                   read_flat_config('nav.conf')
+    :returns: None, but will raise a ConfigurationError if verification failed.
+
+    """
+    if not config:
+        return
+
+    # First, verify presence of absolutely required options
+    for option in ('NAV_USER', 'LOG_DIR', 'PID_DIR'):
+        if option not in config:
+            raise ConfigurationError(
+                'Configuration option {} is missing!'.format(option)
+            )
+
+    # Verify that user exists
+    username = config['NAV_USER']
+    try:
+        pwd.getpwnam(username)
+    except KeyError:
+        raise ConfigurationError('No such user: {}'.format(username))
+
+    # Verify that directories exist and are writable
+    for option, moniker in (('LOG_DIR', 'log'), ('PID_DIR', 'pid file')):
+        dir_name = config[option]
+        if not os.path.isdir(dir_name):
+            raise ConfigurationError(
+                'The {} directory {} does not exist or is not a directory'.format(
+                    moniker, dir_name
+                )
+            )
+        if not _is_directory_writable_by_user(dir_name, username):
+            raise ConfigurationError(
+                'The {} directory {} is not writeable for the {} user'.format(
+                    moniker, dir_name, username
+                )
+            )
+
+
+def _is_directory_writable_by_user(directory, user):
+    dir_stat = os.stat(directory)
+
+    pw = pwd.getpwnam(user)
+    uid, gid = pw.pw_uid, pw.pw_gid
+    dir_mode = dir_stat[stat.ST_MODE]
+
+    # use directory_mode as mask
+    if (
+        uid == dir_stat[stat.ST_UID] and stat.S_IRWXU & dir_mode == stat.S_IRWXU
+    ):  # user is owner and has rwx
+        return True
+    elif (
+        gid == dir_stat[stat.ST_GID] and stat.S_IRWXG & dir_mode == stat.S_IRWXG
+    ):  # user is in group & group has rwx
```

Comment:
what if user is part of multiple groups?

Reply:
Good point.